#! /bin/sh -e
## 5_htmlize-noninteractive.dpatch by Roland Mas <lolando@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixed behaviour of htmlize.el in non-interactive mode

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch -f --no-backup-if-mismatch -p1 < $0;;
       -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
	*)
		echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
		exit 1;;
esac

exit 0
				    
diff -urNad 5.emacs-goodies-el.tmp/elisp/htmlize.el 5.emacs-goodies-el/elisp/htmlize.el
--- 5.emacs-goodies-el.tmp/elisp/htmlize.el	2003-03-19 14:19:30.000000000 +0100
+++ 5.emacs-goodies-el/elisp/htmlize.el	2003-03-19 22:25:26.000000000 +0100
@@ -765,9 +765,13 @@
 	    (funcall ,func ,@args)))))
 
 ;;;###autoload
-(defun htmlize-buffer (&optional buffer)
+(defun htmlize-buffer (&optional buffer no-switch-to-buffer)
   "Convert buffer to HTML, preserving the font-lock colorization.
-HTML contents will be provided in a new buffer."
+HTML contents will be provided in a new buffer.
+
+specifying no-switch-to-buffer to t will not switch the 
+currently displayed buffer to the new buffer.
+"
   (interactive)
   (or buffer
       (setq buffer (current-buffer)))
@@ -785,7 +789,9 @@
 					      (buffer-file-name)))
 					  "*html*"))))
 	 next-change face face-object)
-    (switch-to-buffer newbuf)
+    (if no-switch-to-buffer
+        (set-buffer newbuf)
+      (switch-to-buffer newbuf))
     (buffer-disable-undo)
     (insert (htmlize-method doctype) ?\n
 	    (format "<!-- Created by htmlize-%s in %s mode. -->\n"
